### ISSUE TITLE
Default automaticallyCancelling to false?

### ISSUE THREAD

**Issue by robertmryan** [Alamofire/Alamofire#3727]
* [x] I've read, understood, and done my best to follow the [*CONTRIBUTING guidelines](https://github.com/Alamofire/Alamofire/blob/master/CONTRIBUTING.md).

## Problem

The `automaticallyCancelling` option defaults to `false`.

## Feature Request

It should default to `true`.

## Value to Alamofire

One of the key benefits of Swift concurrency is its first-class support of cancelation. And we have *structured concurrency,* where the cancelation of a parent task will automatically cancel its children. When writing Swift concurrency code, we often strive to ensure cancelation support. In most of Apple's asynchronous API, they offer cancelation support. 

For example, if I want a cancelable request, I understand that it would have to be like:

```swift
func getDataConcurrency(for url: URL) async throws -> SomeModel {
    try await AF.request(url)
        .validate()
        .serializingDecodable(SomeModel.self, automaticallyCancelling: true)
        .value
}
```

I am very confused why you would offer cancelation support, but then default it to being turned off (if that is, indeed, what `automaticallyCancelling` does). I have never seen that as the default behavior in any other asynchronous API. I am hard-pressed to even imagine why someone would want to turn this off. If I cancel some task, of course I would want the request canceled, too. And if one *really* didn't want cancelation support for some reason, we would just wrap it in unstructured concurrency. 

Or have I misinterpreted the purpose of this `automaticallyCancelling` feature? If so, perhaps we can clarify the documentation. 

Bottom line, I would suggest making `automaticallyCancelling` default to `true` (even if that is a breaking change). Defaulting this to `false` seems like it only invites confusion. 

(P.S., I know that "Issues" is not intended for general questions like this, and I apologize, but an inquiry about the author's intent is not going to be accepted on StackOverflow.)

**Comment by jshier** [comment]
This will be short single I'm on vacation, but my thought here was to match the semantics users were used to from Alamofire and `URLSession`. Additionally, since there were significant bugs in the first versions of Swift's concurrency feature around cancellation and its propagation, this was designed around a misunderstanding of what the correct behavior should be. Changing it now to match what `URLSession` does probably makes sense. I'll reevaluate when I get back.

**Comment by robertmryan** [comment]
Yeah, I feel your pain re early backward support for Swift concurrency. I actually completely refactored a project (back in Xcode 13.2, IIRC) and had to unwind it because of the problems you reference. But I think it has subsequently stabilized. 

Thanks for your consideration.

**Comment by jshier** [comment]
@robertmryan It will probably be merged soon, but if you like, take a look at #3757 and see if it behaves the way you expect with automatic cancellation defaulting to `true`.

**Comment by jshier** [comment]
This change has been merged and should go out as part of 5.8 soon.

**Comment by robertmryan** [comment]
> @robertmryan It will probably be merged soon, but if you like, take a look at #3757 and see if it behaves the way you expect with automatic cancellation defaulting to `true`.

Sorry, I've been buried and never got around to reviewing further. My apologies.

**Comment by jshier** [comment]
No problem, feel free to take a look and file an issues you find.
